### PR TITLE
Add parser for NIH intersections shapefile

### PR DESCRIPTION
**Pull request by flibbertigibbet**
We will have two shapefiles to parse, one with additional information on street segments, and the other with additional intersection information.  This adds a parser for the intersections shapefile that locates the intersections previously imported from OpenStreetMap.

### CODE REVIEW

File: src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileIntersectionGraphBuilderImpl.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        }
+        it2.close();
+        it2 = null;
+
+        LOG.info("Have {} NIH intersections to process...", featureList.size());
+
+        int foundIntersectionCt = 0;
+        int missingIntersectionCt = 0;
+
+        for (SimpleFeature feature : featureList) {
```
Comment (ddohler) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileIntersectionGraphBuilderImpl.java:123]: Is it possible to combine this loop and the above `while` loop? I.e.

``` java
FeatureIterator<SimpleFeature> it2 = features.features();
while (it2.hasNext()) {
    SimpleFeature feature = it2.next();
    if (feature.getDefaultGeometry() == null) {
    ...
}
it2.close();
it2 = null;
```

**Comment by ddohler** [comment]
Looks good so far.

File: src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileIntersectionGraphBuilderImpl.java (hunk 1 above)
Comment (flibbertigibbet) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileIntersectionGraphBuilderImpl.java:123]: I'm not sure why the original shapefile importer processes things that way, in two loops, apart from possibly isolating potential points of failure.  Yes, I think both importers could be handled in a single loop.

Comment (flibbertigibbet) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileIntersectionGraphBuilderImpl.java:123]: I think I will leave it this way for now, partly to isolate potential issues in the file-processing step, so the file can be closed sooner than later, and partly so we can get the full feature count before processing them, so we can check if any went missing.

**Pull request merged**